src/corner_kick/README.md

```diff
@@ -36,20 +53,60 @@ The project is defined as follow:
 -   `config` - contains webpack (build tool) configuration files
 -   `launch` - contains roslaunch file
 -   `src`
-    -   `main` - contains code for the main process
-    -   `renderer` - contains code for the rendering process
-        -   `components`
-            -   `containers` - contains React stateful components
-            -   `ui` - contains repeatable UI components
-        -   `modules` - contains high-level UI components (logger, visualizer, etc.)
-    -   `shared` - contains code shared by the two process
-        -   `ros` - contains the service communicating with ROS
-            -   `types` - contains ROS message type schema
+    -   `assets` - contains image assets
+    -   `components` - contains repeatable UI components
+    -   `containers` - contains high-level UI components (logger, visualizer, etc.)
+    -   `constants` - contains fixed values used throughout the application
+    -   `store` - contains logic for the application Redux state
```

Comment:
Could we add a sub-entry here for `reducers` with a quick description and this link: https://redux.js.org/basics/reducers?

Reply:
#### Reducer functions and Redux saga

Reducer functions are the cornerstone of the Redux approach. Once a Redux action is dispatched, reducer functions
determine what the new state will look like as a result of the action sent. Reducer functions do not permit for
any side-effects to ensure a deterministic behaviour. For a state and action, the same new state should be returned
by the reducer funtion. This, however, does not permit for more complex logic, such as communicating with ROS,
to occur. To allow for this, we introduce sagas, which can intercept and act on Redux action. They can conduct some
business logic work, and reflect the result of this work in the state by emmiting a Redux action when needed.

For more information about reducer functions, read the [official Redux doc on that topic](https://redux.js.org/basics/reducers).
For more information about Redux sagas, read the [official Redux saga doc](https://redux-saga.js.org/).